From 0693cc5c2c588acfe4260ae30886fac8f55d4f2d Mon Sep 17 00:00:00 2001
From: "duncan.macleod" <duncan.macleod@ligo.org>
Date: Tue, 20 Dec 2022 10:16:01 +0000
Subject: [PATCH] bottle: handle getargspec for Python 3.11

patch bottle.py to handle removal of `inspect.getargspec`, closes #128,
see https://github.com/bottlepy/bottle/commit/fd78d672ebb362f0e5116ad7c0057da114b25b40
---
 gstlal/python/bottle.py | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/gstlal/python/bottle.py b/gstlal/python/bottle.py
index fcd659320..ddcf524cd 100644
--- a/gstlal/python/bottle.py
+++ b/gstlal/python/bottle.py
@@ -41,8 +41,15 @@ import base64, cgi, email.utils, functools, hmac, imp, itertools, mimetypes,\
 from datetime import date as datedate, datetime, timedelta
 from tempfile import TemporaryFile
 from traceback import format_exc, print_exc
-from inspect import getargspec
 from unicodedata import normalize
+try:
+    from inspect import getargspec
+except ImportError:  # Python >= 3.11
+    from inspect import getfullargspec
+    def getargspec(func):
+        spec = getfullargspec(func)
+        kwargs = makelist(spec[0]) + makelist(spec.kwonlyargs)
+        return kwargs, spec[1], spec[2], spec[3]
 
 
 try: from simplejson import dumps as json_dumps, loads as json_lds
-- 
2.30.2

